### ISSUE TITLE
Lineplot - rethink default binwidth of 1 when "not binning"

### ISSUE THREAD

**Issue by bobular** [VEuPathDB/web-monorepo#457]
At the moment, the default behaviour of the lineplot client for a continuous variable _when the user has not asked for any binning_, is to ask the back end to use a binwidth of 1.

I know this was a decision made for date variables (bin by 1 day, which seems sensible) but I'm not sure it's a great idea for number variables.

In most cases it's fine, but if a variable has a `max-min < 1` then the bin width of 1 will lump everything together. If the user toggles binning on, then the metadata-suggested binwidth (smaller than 1) will be used, and the binning should make much more sense.

@danicahelb @nkittur-uga do you know of any variables with small ranges (e.g. 0 to 1) that would be helpful exploring this issue? Please could you also triage this ticket?

**Comment by dmfalke** [comment]
I have a very vague recollection that the backend would figure out an appropriate default bin, if one isn't specified in the request. I haven't tested this, but assuming this is the case, then I think we ought to be able to remove the bin spec if the user hasn't asked for any. If the backend isn't doing that, perhaps it should?

**Comment by nkittur-uga** [comment]
@bobular [Here's one variable](https://clinepidb.org/ce/app/workspace/analyses/DS_51b40fe2e2/new/variables/EUPATH_0000609/EUPATH_0020243) that may be useful for testing: PRISM2 ICEMR Cohort study; variable "Mosquitoes with oocysts proportion" 

[Here's another variable](https://clinepidb.org/ce/app/workspace/analyses/DS_8786631aaf/new/variables/EUPATH_0000096/EUPATH_0042239): PROMOTE Birth Cohort study; variable "Proportion of Plasmodium positive blood smears during pregnancy, by microscopy"

**Comment by bobular** [comment]
@dmfalke 
I think it can do auto binwidth. But we also know from study metadata what the appropriate bin width is. But the main point is that binning is off by default. For dates it makes a lot of sense to bin by one day. For numeric continuous, I'm not sure why we chose to bin by 1.

**Comment by bobular** [comment]
Thanks @nkittur-uga - now I can show an example of what a lineplot with one of these variables on the x-axis looks like by default (binning toggle is off; but binning with binwidth=1 still requested). And below that, a more sensible plot of the same variables with the binning toggle on (it uses the metadata-recommended binwidth of 0.1).

![image](https://github.com/VEuPathDB/web-monorepo/assets/308639/5652419c-b22a-4491-938c-531afb7176a3)

![image](https://github.com/VEuPathDB/web-monorepo/assets/308639/ed7a695f-f57e-4942-8fa9-94dc85fe41dc)

**Comment by nkittur-uga** [comment]
Ooh I LIKE! So much better.

**Comment by bobular** [comment]
Yes, but we have to ask, why did we decide to do binning with binwidth=1 when the user has asked for "no binning" via the toggle?

It makes sense that we bin by "1 day" for date variables, but not for regular number variables.

Perhaps binning should be always-on for continuous x-axis variables?

**Comment by bobular** [comment]
Though if a continuous variable has a small `distinctValuesCount` it could possibly be treated as a categorical axis, like I think we do for overlay variables.
